kotlin-native/runtime/src/legacymm/cpp/Memory.cpp

```diff
@@ -253,7 +253,7 @@ class CycleDetector : private kotlin::Pinned, public KonanAllocatorAware {
 // TODO: can we pass this variable as an explicit argument?
 THREAD_LOCAL_VARIABLE MemoryState* memoryState = nullptr;
 
-#ifdef KONAN_WINDOWS
+#if defined(KONAN_WINDOWS) && !defined(KONAN_DISABLE_CUSTOM_WIN32_TLS)
```

Comment:
Thinking about this. This file is compiled once per platform and distributed, right? Even if the user sets that define, that won't change its behaviour.

Reply:
Exactly.